plugins/org.polymap.kaps/src/org/polymap/kaps/ui/form/VertragsdatenBaulandGrunddatenFormEditorPage.java

```diff
@@ -167,6 +159,24 @@ public void run() {
                 .setLayoutData( right().top( lastLine ).create() ).setParent( client ).create();
 
         lastLine = newLine;
+        
+        Double tatsächlichesBaujahr = null;
+        Double bereinigtesBaujahr = null;
+        NHK2010BewertungComposite bewertung2010 = NHK2010BewertungComposite.Mixin.forVertrag(vb.vertrag().get());
+        if(bewertung2010 != null) {
+        	Iterable<NHK2010BewertungGebaeudeComposite> bewertungenGebaeude2010 = NHK2010BewertungGebaeudeComposite.Mixin.forBewertung(bewertung2010);
+        	for(NHK2010BewertungGebaeudeComposite bewertungGebaeude2010 : bewertungenGebaeude2010) {
+        		tatsächlichesBaujahr = bewertungGebaeude2010.tatsaechlichesBaujahr().get();
+        		bereinigtesBaujahr = bewertungGebaeude2010.bereinigtesBaujahr().get();
+        	}
+        }
+        if(bereinigtesBaujahr != null) {
+        	vb.baujahrBereinigt().set(bereinigtesBaujahr.intValue());
```

Comment:
Hier ebenfalls, Berechnung auslagern in separate Methode und nicht am Entity speichern sondern mit *site.setFieldValue()*... setzen.